sdk/keyvault/platform-matrix.json

```diff
@@ -0,0 +1,13 @@
+{
+  "matrix": {
+      "Groups": {
+          "ubuntu-18.04_HSM": {
+              "OSVmImage": "MMSUbuntu18.04", "Pool": "azsdk-pool-mms-ubuntu-1804-general",
+              "ArmTemplateParameters": "@{ enableHsm = $true }"
+          },
+          "macOS-10.15": { "OSVmImage": "macOS-10.15", "Pool": "Azure Pipelines" },
+          "windows-2019": { "OSVmImage": "MMS2019", "Pool": "azsdk-pool-mms-win-2019-general" }
+      },
+      "TestTargetFramework": [ "netcoreapp2.1", "netcoreapp3.1", "net47" ]
```

Comment:
net74 only works on Windows, which maybe is the problem you Teams'd me about yesterday.

Reply:
Since the matrix is included as sparse, it will only output net47 for windows right now. Perhaps I should add an exclude for `!windows` in case anyone ever tries to generate the full matrix instead of the sparse one.

Reply:
I guess I still don't get the sparse and other option. Where were the docs again?

Reply:
https://github.com/Azure/azure-sdk-tools/blob/master/eng/common/scripts/job-matrix/README.md#sparse

Reply:
Think about sparse as slicing a line through the middle of the matrix, so if we had a simple 3x3 square matrix, the sparse output would be: `[ column(0), row(0) ], [ column(1), row(1), column(2), row(2) ]`. For non-square matrices, it's the same thing, just with modulus, e.g. for a 3x2x1 matrix: `[ vector1(i % vector1.Length-1), vector2(i % vector2.Length-1), vector3(i % vector3.Length-1) ]`

Reply:
So in the case of this config, that ends up being a sparse output of: ubuntu+netcoreapp2.1, macOS+netcoreapp3.1, windows+net47. If we were to generate the matrix with a selection "all" instead of "sparse" then it would give us the entire combination set (9 elements) as opposed to 3 elements.

Reply:
So...if we anticipate the matrix being selected with "all", we can add an exclude that says "remove ubuntu/mac + net47 from the output"